homeassistant/components/myuplink/config_flow.py

```diff
@@ -23,3 +28,32 @@ def logger(self) -> logging.Logger:
     def extra_authorize_data(self) -> dict[str, Any]:
         """Extra data that needs to be appended to the authorize url."""
         return {"scope": " ".join(OAUTH2_SCOPES)}
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self.config_entry_reauth = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: Mapping[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+            )
+        return await self.async_step_user()
+
+    async def async_oauth_create_entry(self, data: dict) -> FlowResult:
+        """Create or update the config entry."""
+        if self.config_entry_reauth:
+            self.hass.config_entries.async_update_entry(
```

Comment:
We have a helper now to do this:

https://github.com/home-assistant/core/blob/d6efdc47a5a8cb06ea00cba9d36e6b376d97b467/homeassistant/config_entries.py#L2030-L2041